File: demo_client.php
<?php
$BASE = "http://localhost:3000";

function call($method, $url, $body=null, $query=[]) {
  $ch = curl_init();
  if (!empty($query)) $url .= '?' . http_build_query($query);
  curl_setopt_array($ch, [
    CURLOPT_URL => $url,
    CURLOPT_CUSTOMREQUEST => $method,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_HTTPHEADER => ['Content-Type: application/json'],
    CURLOPT_POSTFIELDS => $body ? json_encode($body) : null
  ]);
  $res = curl_exec($ch);
  if ($res === false) die(curl_error($ch));
  $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
  curl_close($ch);
  if ($status >= 400) die("HTTP $status: $res\n");
  return json_decode($res, true);
}

function pretty($title, $data) {
  echo "\n=== $title ===\n";
  echo json_encode($data, JSON_PRETTY_PRINT) . "\n";
}

// 1) Create space
$space = call('POST', "$BASE/spaces", [
  "name"=>"Main Hall L1", "location"=>"Campus A - Tower 1", "totalSeats"=>5
]);
pretty("Create Space", $space);
$spaceId = $space["id"];

// 2) Seed seats
pretty("Seed Seats", call('POST', "$BASE/spaces/$spaceId/seed"));

// 3) List AVAILABLE
$seats = call('GET', "$BASE/spaces/$spaceId/seats", null, ["status"=>"AVAILABLE"]);
pretty("List AVAILABLE Seats", $seats);
$seatId = $seats[0]["id"];

// 4) Lock
pretty("Lock Seat", call('POST', "$BASE/spaces/$spaceId/seats/$seatId/lock"));

// 5) Set RESERVED
pretty("Set RESERVED", call('PATCH', "$BASE/spaces/$spaceId/seats/$seatId/status", ["status"=>"RESERVED"]));

// 6) List RESERVED
pretty("List RESERVED Seats", call('GET', "$BASE/spaces/$spaceId/seats", null, ["status"=>"RESERVED"]));

echo "\n✅ Demo completed.\n";

// To run this demo, ensure the server is running on localhost:3000
// Then execute: & "C:\xampp\php\php.exe" .\demo_client.php

File: demo_client.py
import requests, json, sys

BASE = "http://localhost:3000"

def pretty(title, data):
    print(f"\n=== {title} ===")
    print(json.dumps(data, indent=2))

# 1) Create space
r = requests.post(f"{BASE}/spaces", json={
    "name":"Main Hall L1", "location":"Campus A - Tower 1", "totalSeats": 5
})
r.raise_for_status()
space = r.json()
pretty("Create Space", space)
space_id = space["id"]

# 2) Seed seats S1..S5
r = requests.post(f"{BASE}/spaces/{space_id}/seed")
r.raise_for_status()
pretty("Seed Seats", r.json())

# 3) List AVAILABLE seats
r = requests.get(f"{BASE}/spaces/{space_id}/seats", params={"status":"AVAILABLE"})
r.raise_for_status()
seats = r.json()
pretty("List AVAILABLE Seats", seats)
seat_id = seats[0]["id"]

# 4) Lock (hold) a seat
r = requests.post(f"{BASE}/spaces/{space_id}/seats/{seat_id}/lock")
r.raise_for_status()
pretty(f"Lock Seat {seat_id}", r.json())

# 5) Mark as RESERVED (checkout)
r = requests.patch(f"{BASE}/spaces/{space_id}/seats/{seat_id}/status", json={"status":"RESERVED"})
r.raise_for_status()
pretty(f"Set Seat {seat_id} to RESERVED", r.json())

# 6) List RESERVED seats
r = requests.get(f"{BASE}/spaces/{space_id}/seats", params={"status":"RESERVED"})
r.raise_for_status()
pretty("List RESERVED Seats", r.json())

print("\n✅ Demo completed.")

# To run this demo, ensure the server is running on localhost:3000
# Then execute: python demo_client.py